### ISSUE TITLE
How to handle Java 18 disabling of SecurityManager?

### ISSUE THREAD

**Issue by jhanders34** [OpenLiberty/open-liberty#20289]
**Associated Work Item(s):** https://github.com/OpenLiberty/open-liberty/issues/18614

Java 18 continues the deprecation of SecurityManager by changing the default to not allow System.setSecurityManager to be called.  In order for you to be able to override this behavior, you need to set a system property as part of the JVM invocation as an argument as -Djava.security.manager=allow.  Without that, Java security will not be able to be enabled.  You cannot set this property dynamically during the runtime if we see the bootstrap.properties setting to enable Java security.  

As such, we were going down the path of requiring customers to set both the bootstrap property and the system property if using Java 18 or higher.  Alasdair brought up the other option of just not supporting Java security with Java 18 and above since likely that is what is going to happen anyway in Java 21.

Options to talk about:

1. Require customers to set both properties and have a message telling them such if they don't set -Djava.security.manager=allow
2. Update server script to be able to tell what version of Java is being used and if it is Java 18 and above set the system property for the customer if the bootstrap property enabling Java security is set.
3. No longer support Java security with Java 18 and above Java versions meaning we would ignore the bootstrap property and put out a message indicating we ignored it and didn't enable Java security.

#1 is what is currently in the code except for the message.  #2 is not something Alasdair wants to do to run another program in server script just to get the Java version.  For #3, is there any concern about Java / Jakarta EE not working fully with Java 18 if Java security is not able to be enabled?

**Comment by jhanders34** [comment]
The decision was to go with #3.  Also start a conversation with the community for Jakarta EE 10 TCK on how they plan to handle Java 18 and higher having Java security disabled by default in the JDK.

**Comment by sgammon** [comment]
@jhanders34 Hi! You don't know me, and Google brought me here on a completely unrelated bug mission. But nonetheless I copied-and-pasted this from your PR:

```
-Djava.security.manager=allowed
```

It didn't work! I dug and dug and couldn't find why, until I realized [it should be](https://github.com/openjdk/jdk21/blob/master/src/java.base/share/classes/java/lang/SecurityManager.java#L151-L156):

```
-Djava.security.manager=allow
```

I have no idea if you'll get this comment but it would be awesome to update your PR description so that users aren't misguided 😅  even though it's not your fault at all.
